### ISSUE TITLE
Unsupported objectType

### ISSUE THREAD

**Issue by Alienuser** [electricbolt/dart-propertylistserialization#2]
Hello together,

I'm using the library for my Flutter app. I saved a `[Double]` with iOS as an **plist00** in Core Data. 
Now I try to receive this data in my Flutter app.

`String.fromCharCodes(tableColumn)` gives me the following:
`bplist00Ô\^A\^B\^C\^D\^E\^F\^G
X$versionY$archiverT$topX$objects\^R
`

As far as I can see I have a valid **plist00**, therefore the processing is running fine. I'm doing it with:
`Uint8List columnData = Uint8List.fromList(element["ZSHOOTS"]);
var bytes = ByteData.sublistView(columnData);
PropertyListSerialization.propertyListWithData(bytes) as List;`

 At some point I'm getting the following error:
`error PropertyListReadStreamException: Unsupported operation: Unsupported plist objectType 8`

While debugging I noticed in the binarypropertylistreader.dart file, that there is no case for 8.
It goes until 6 and continues with A. Is there a reason for this?

The `dict` variable in line 126 contains 2 entries before the crash
0: "$version" -> 100000
1: "$archiver" -> "NSKeyedArchiver"

I assume the problem is the ** topX** part? 

Thanks!

**Comment by Alienuser** [comment]
My binary .plist file ist the following. Its archived as a zip. Please export it first.
[test.plist.zip](https://github.com/electricbolt/dart-propertylistserialization/files/8069364/test.plist.zip)
.

**Comment by electricbolt** [comment]
Hi, the test.plist.zip above is a textual plist. When I unzipped I have the following:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>$archiver</key>
	<string>NSKeyedArchiver</string>
	<key>$objects</key>
	<array>
		<string>$null</string>
		<dict>
			<key>$class</key>
			<dict>
				<key>CF$UID</key>
				<integer>4</integer>
			</dict>
			<key>NS.objects</key>
			<array>
				<dict>
					<key>CF$UID</key>
					<integer>2</integer>
				</dict>
				<dict>
					<key>CF$UID</key>
					<integer>3</integer>
				</dict>
			</array>
		</dict>
		<real>15</real>
		<real>16</real>
		<dict>
			<key>$classes</key>
			<array>
				<string>NSArray</string>
				<string>NSObject</string>
			</array>
			<key>$classname</key>
			<string>NSArray</string>
		</dict>
	</array>
	<key>$top</key>
	<dict>
		<key>root</key>
		<dict>
			<key>CF$UID</key>
			<integer>1</integer>
		</dict>
	</dict>
	<key>$version</key>
	<integer>100000</integer>
</dict>
</plist>

```

**Comment by electricbolt** [comment]
Ah my apologies dragging into my text editor automatically converted it into textual format. I'll take a look at the issue now.

Thanks.

**Comment by electricbolt** [comment]
It appears you're attempting to deserialize a NSKeyedArchiver, which is a _superset_ of a property list, as property lists don't have encodings for NSSet, NSOrderedSet, NSUUID and NSURL. I _don't_ currently have any plans to extend the dart-propertylistserialization library to support these additional NSKeyedArchiver object types. 

**Suggested workaround**

Use the iOS native property list serialization APIs to convert from binary to XML format (e.g. this will convert the binary NSUUID data type to an XML integer stored in a dictionary under the key "CF$UID"), then pass the XML to flutter, at which point you could successfully parse the XML using the dart-propertylistserialization library.

Thanks.

**Comment by Alienuser** [comment]
Thanks for letting me know.

**Comment by electricbolt** [comment]
Hi Alienuser.

I've taken a look at what's required to support the NSKeyedArchiver CF$UID (type 0x8), and it only turned out to be a few lines of code + unit test. I will only support reading binary plist's for now, and it would allow you to read the attached test.plist file without issues.

Is this all you need? If so I'll package it up and publish.

**Comment by Alienuser** [comment]
Hi electricbolt, that sounds nice. Thank you very much!

**Comment by electricbolt** [comment]
Hi @Alienuser. I've just published 1.2.0 to pub.dev. To enable parsing of CF$UID constructs add the optional `keyedArchive` parameter:

`var graph = PropertyListSerialization.propertyListWithData(bytes, keyedArchive: true) as List;`

Thanks.
